components/crypto/src/utils.rs

```diff
@@ -138,6 +122,124 @@ impl EncryptedKeys {
     }
 }
 
+#[derive(Debug, Serialize, Deserialize, Clone, Eq, PartialEq)]
+pub struct Keys {
+    pub consensus_pk: PublicKey,
+    pub consensus_sk: SecretKey,
+    pub service_pk: PublicKey,
+    pub service_sk: SecretKey,
+    pub identity_pk: kx::PublicKey,
+    pub identity_sk: kx::SecretKey,
+}
+
+pub fn save_master_key<P: AsRef<Path>, W: AsRef<[u8]>>(
+    path: P,
+    pass_phrase: W,
+    key: &secret_tree::Seed,
+) -> Result<(), Error> {
+    let encrypted_key = EncryptedMasterKey::encrypt(key, pass_phrase)?;
+    let file_content =
+        toml::to_string_pretty(&encrypted_key).map_err(|e| Error::new(ErrorKind::Other, e))?;
+    let mut open_options = OpenOptions::new();
+    open_options.create(true).write(true);
+    #[cfg(unix)]
+    open_options.mode(0o_600);
+    let mut file = open_options.open(path.as_ref())?;
+    file.write_all(file_content.as_bytes())?;
+
+    Ok(())
+}
+
+#[derive(Serialize, Deserialize)]
+struct EncryptedMasterKey {
+    key: ErasedPwBox,
+}
+
+impl EncryptedMasterKey {
+    fn encrypt(
+        key: &secret_tree::Seed,
+        pass_phrase: impl AsRef<[u8]>,
+    ) -> Result<EncryptedMasterKey, Error> {
+        let mut rng = thread_rng();
+        let mut eraser = Eraser::new();
+        eraser.add_suite::<Sodium>();
+        let pwbox = Sodium::build_box(&mut rng)
+            .seal(pass_phrase, key)
+            .map_err(|_| Error::new(ErrorKind::Other, "Couldn't create a pw box"))?;
+        let encrypted_key = eraser
+            .erase(&pwbox)
+            .map_err(|_| Error::new(ErrorKind::Other, "Couldn't convert a pw box"))?;
+
+        Ok(EncryptedMasterKey { key: encrypted_key })
+    }
+
+    fn decrypt(self, pass_phrase: impl AsRef<[u8]>) -> Result<secret_tree::Seed, Error> {
+        let mut eraser = Eraser::new();
+        eraser.add_suite::<Sodium>();
+        let restored = eraser
+            .restore(&self.key)
+            .map_err(|_| Error::new(ErrorKind::Other, "Couldn't restore a secret key"))?;
+        assert_eq!(restored.len(), SEED_LENGTH);
+        let seed_bytes = restored
+            .open(pass_phrase)
+            .map_err(|_| Error::new(ErrorKind::Other, "Couldn't open an encrypted key"))?;
+
+        let mut seed: [u8; 32] = [0; 32];
+        seed.copy_from_slice(&seed_bytes[..]);
+        Ok(seed)
+    }
+}
+
+pub fn generate_keys<P: AsRef<Path>>(path: P, passphrase: &[u8]) -> Keys {
+    let tree = SecretTree::new(&mut thread_rng());
+    save_master_key(path, passphrase, tree.seed()).expect("Error generating master key.");
+    generate_keys_from_master_password(tree)
+}
+
+fn generate_keys_from_master_password(tree: SecretTree) -> Keys {
```

Comment:
Why don't you return `Result<Keys, Error` here ? It would let you to get rid of unwraps.